zk_toolbox/crates/zk_supervisor/src/commands/test/rust.rs

```diff
@@ -1,32 +1,73 @@
+use std::{
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+
 use anyhow::Context;
 use common::{cmd::Cmd, config::global_config, db::wait_for_db, logger};
 use config::EcosystemConfig;
+use url::Url;
 use xshell::{cmd, Shell};
 
 use super::args::rust::RustArgs;
 use crate::{
     commands::database,
-    dals::get_test_dals,
+    dals::{Dal, CORE_DAL_PATH, PROVER_DAL_PATH},
     messages::{
         MSG_CARGO_NEXTEST_MISSING_ERR, MSG_CHAIN_NOT_FOUND_ERR, MSG_POSTGRES_CONFIG_NOT_FOUND_ERR,
         MSG_RESETTING_TEST_DATABASES, MSG_UNIT_TESTS_RUN_SUCCESS, MSG_USING_CARGO_NEXTEST,
     },
 };
 
 pub async fn run(shell: &Shell, args: RustArgs) -> anyhow::Result<()> {
-    let ecosystem = EcosystemConfig::from_file(shell)?;
-    let chain = ecosystem
-        .clone()
-        .load_chain(global_config().chain_name.clone())
-        .context(MSG_CHAIN_NOT_FOUND_ERR)?;
-    let general_config = chain.get_general_config()?;
-    let postgres = general_config
-        .postgres_config
-        .context(MSG_POSTGRES_CONFIG_NOT_FOUND_ERR)?;
+    let link_to_code = if let Some(link_to_code) = args.link_to_code {
+        PathBuf::from(link_to_code)
+    } else {
+        EcosystemConfig::from_file(shell)?.link_to_code
+    };
```

Comment:
do we really need it? 
Because ecosystem always exists

Reply:
True! We only needed the dbs, i will change this